syntax = "proto3";

package bookservice;


message Book {
    int64 id = 1;
    string title = 2;
    string author = 3;
    string description = 4;
    int32 publication_year = 5;
}

message AddBookRequest {
    Book book = 1;
}

message AddBookResponse {
    string status = 1;
    Book book = 2;
}

message GetBookRequest {
    int64 id = 1;
}

message GetBookResponse {
    Book book = 1;
}

message UpdateBookRequest {
    Book book = 1;
}

message UpdateBookResponse {
    string status = 1;
    Book book = 2;
}

message DeleteBookRequest {
    int32 id = 1;
}

message DeleteBookResponse {
    string status = 1;
}

service BookService {
    rpc AddBook(AddBookRequest) returns (AddBookResponse);
    rpc GetBook(GetBookRequest) returns (GetBookResponse);
    rpc UpdateBook(UpdateBookRequest) returns (UpdateBookResponse);
    rpc DeleteBook(DeleteBookRequest) returns (DeleteBookResponse);
}
